include/z64actor.h

```diff
@@ -532,4 +532,31 @@ typedef enum {
 #define UPDBGCHECKINFO_FLAG_6 (1 << 6) // disable water ripples
 #define UPDBGCHECKINFO_FLAG_7 (1 << 7) // alternate wall check?
 
+typedef enum {
+    /* 0x0 */ NPC_TALK_STATE_IDLE, // NPC not currently talking to player
+    /* 0x1 */ NPC_TALK_STATE_TALKING, // NPC is currently talking to player
+    /* 0x2 */ NPC_TALK_STATE_ACTION, // An NPC-defined action triggered in the conversation
+    /* 0x3 */ NPC_TALK_STATE_ITEM_GIVEN // NPC finished giving an item and text box is done
+} NpcTalkState;
+
+typedef enum {
+    /* 0x0 */ NPC_PLAYER_TRACKING_AUTO_TURN,  // The actor will periodically turn to face the player
+    /* 0x1 */ NPC_PLAYER_TRACKING_NONE, // Don't track the player
+    /* 0x2 */ NPC_PLAYER_TRACKING_HEAD_AND_TORSO, // Track player by turning the head and the torso
+    /* 0x3 */ NPC_PLAYER_TRACKING_HEAD, // Track player by turning the head
+    /* 0x4 */ NPC_PLAYER_TRACKING_FULL_BODY // Track player by turning the body, torso and head
+} NpcPlayerTrackingOption;
+
+typedef struct {
+    /* 0x00 */ s16 talkState;
+    /* 0x02 */ s16 playerTrackingOpt;
+    /* 0x04 */ s16 autoTurnTimer;
+    /* 0x06 */ s16 autoTurnState;
+    /* 0x08 */ Vec3s rotHead;
+    /* 0x0E */ Vec3s rotTorso;
+    /* 0x14 */ f32 yPosOffset; // Y position offset to add to actor position when calculating angle to player
+    /* 0x18 */ Vec3f playerPosition;
```

Comment:
Ther are *very* niche cases where this isn't exactly the player position (but not far from it really), i.e.
- ingo, mido, kokiris can track the camera position `playerPosition = play->view.eye`
- zoras tracking the player's position but not along y (???) `playerPosition.y = this->actor.world.pos.y;`
- zl3 (some zelda version) with `func_80B57D80` tracking a point on a path (idk what's going on there)

would it be doable to change this to trackPosition or something like that? Or is there something I missed that makes this need to be tied to the player besides how it's used

Reply:
I guess `trackPosition` would be better, yeah. The only thing that kind of assumes that this is used to track the player is the `Npc_UpdateAutoTurn` function where the `talkState` is used to check if the player is talking to the player. But those actors that you listed don't even end up there because they don't use the auto-turning mode.

So yeah, changing this to not be player-specific is definitely better. What would be the best name here? `trackTarget`? `trackPosition`? `trackPoint`?

Reply:
I would say `trackPos` is fine

Reply:
What about all the other places where "player tracking" is mentioned? `NpcPlayerTrackingMode` -> `NpcTrackingMode`? `Npc_TrackPlayer` -> `Npc_TrackPoint`? Should this be generalized everywhere even though the majority of the uses are just player tracking?

Reply:
yeah I think "(npc) tracking" is enough, no need to really specify player?

except for `Npc_UpdateAutoTurn` and `NPC_PLAYER_TRACKING_AUTO_TURN` since indeed the check for talkState inside it doesn't make sense in general (if the trackPos isn't necessarily player)
i.e. maybe `NPC_PLAYER_TRACKING_` -> `NPC_TRACKING_` for the enum prefix, but `NPC_PLAYER_TRACKING_AUTO_TURN` -> `NPC_TRACKING_PLAYER_AUTO_TURN`